src/Sylius/Component/Core/Resolver/ShippableCountriesResolver.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Resolver;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Resolver\ShippableCountriesResolverInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+class ShippableCountriesResolver implements ShippableCountriesResolverInterface
+{
+    /**
+     * @var RepositoryInterface
+     */
+    protected $countryRepository;
+
+    /**
+     * @var ChannelContextInterface
+     */
+    protected $channelContext;
+
+    /**
+     * ShippableCountriesResolver constructor.
+     *
+     * @param RepositoryInterface $countryRepository
+     * @param ChannelContextInterface $channelContext
+     */
+    public function __construct(RepositoryInterface $countryRepository, ChannelContextInterface $channelContext)
+    {
+        $this->countryRepository = $countryRepository;
+        $this->channelContext = $channelContext;
+    }
+
+    /**
+     * @param ChannelInterface|null $channel
+     *
+     * @return array
+     */
+    public function getShippableCountries(ChannelInterface $channel = null): array
+    {
+        if ($channel === null) {
+            $channel = $this->channelContext->getChannel();
+        }
+
+        $countries = $channel->getShippableCountries();
+
+        if ($countries->isEmpty()) {
+            $countries = $this->countryRepository->findAll();
```

Comment:
Yes, this is desired. @pjedrzejewski wanted to have all countries returned if no shipping countries where selected to avoid BC breaks. https://github.com/Sylius/Sylius/issues/9122